hook/hook.go

```diff
@@ -8,15 +8,73 @@
 package hook
 
 import (
+	"math"
+
 	"github.com/ava-labs/avalanchego/vms/components/gas"
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core"
 	"github.com/ava-labs/libevm/core/state"
 	"github.com/ava-labs/libevm/core/types"
 	"github.com/ava-labs/libevm/params"
+	"github.com/holiman/uint256"
 
 	"github.com/ava-labs/strevm/intmath"
 	saeparams "github.com/ava-labs/strevm/params"
 )
 
+// AccountDebit includes an amount that an account should have debited,
+// along with the nonce used to debit the account.
+type AccountDebit struct {
+	Nonce  uint64
+	Amount uint256.Int
+}
+
+// Op is an operation that can be applied to state during the execution of a
+// block.
+type Op struct {
+	// Gas consumed by this operation
+	Gas gas.Gas
+	// GasPrice this operation is willing to spend
+	GasPrice uint256.Int
+	// From specifies the set of accounts and the authorization of funds to be
+	// removed from the accounts.
+	From map[common.Address]AccountDebit
+	// To specifies the amount to increase account balances by. These funds are
+	// not necessarily tied to the funds consumed in the From field. The sum of
+	// the To amounts may even exceed the sum of the From amounts.
+	To map[common.Address]uint256.Int
+}
+
+// ApplyTo applies the operation to the statedb.
+//
+// If an account has insufficient funds, [core.ErrInsufficientFunds] is returned
+// and the statedb is unchanged.
+func (o *Op) ApplyTo(stateDB *state.StateDB) error {
+	for from, acc := range o.From {
+		if balance := stateDB.GetBalance(from); balance.Lt(&acc.Amount) {
```

Comment:
```suggestion
		if b := stateDB.GetBalance(from); b.Lt(&acc.Amount) {
```

(nit, style) The full word only adds noise without any signal as it's made redundant by the name of the method on `StateDB`. [Background](https://google.github.io/styleguide/go/decisions#:~:text=The%20general%20rule%20of%20thumb%20is%20that%20the%20length%20of%20a%20name%20should%20be%20proportional%20to%20the%20size%20of%20its%20scope%20and%20inversely%20proportional%20to%20the%20number%20of%20times%20that%20it%20is%20used%20within%20that%20scope.).